1000$

```diff
@@ -2,61 +2,82 @@
 #include "balance.cpp"
 #include "randomEngine.cpp"
 
+struct Character{
+    int humanNumber1;
+    int humanNumber2;
+    int pcValue1;
+    int pcValue2;
+};
+
 class NumberGame{
-    int computerChoice;
     int randNumber;
-    int humanChoice;
     Balance balance = Balance(1000, 1000);
-    RandomEngine randomService;
+    RandomEngine randomEngine;
 public:
     void run(){
-        humanChoice = getPlayerInput();
-        computerChoice = calcComputerTurn(humanChoice);
-        randNumber = randomService.getRandomNumber();
-            int absComputer = std::abs(computerChoice - randNumber);
-            int absHuman = std::abs(humanChoice - randNumber);
+        while(balance.getHumanScoreBalance() >= 0 || balance.getPcScoreBalance() >= 0) {
+            std::cout << balance.getHumanScoreBalance();
+            Character ch;
+            getPlayerInput(ch);
+            calcComputerTurn(ch);
+            randNumber = randomEngine.getRandomNumber();
+
+            int absComputer = std::abs(ch.pcValue1 + ch.pcValue2 - randNumber);
+            //std::cout << absComputer1 << " " << absComputer2 << " " << absComputerMin << std::endl;
+            int absHuman = std::abs(ch.humanNumber1 + ch.humanNumber2 - randNumber);
+            //std::cout << absHuman1 << " " << absHuman2 << " " << absHumanMin;
+
             std::cout << "Game begin" << std::endl
-                      << "Pc chose: " << computerChoice << std::endl
-                      << "Your choice: " << humanChoice << std::endl
+                      << "Pc chose: " << ch.pcValue1 << " and " << ch.pcValue2 << std::endl
+                      << "Your choice: " << ch.humanNumber1 << " and " << ch.humanNumber2 << std::endl
                       << "Number: " << randNumber << std::endl;
+
             if (absComputer == absHuman) {
-                defineWinnerByRange(humanChoice, computerChoice);
+                defineWinnerByRange(ch);
             } else {
-                defineWinnerByAbs(absHuman, absComputer);
+                defineWinnerByAbs(absComputer, absHuman);
             }
             balance.showBalance();
+        }
     }
-    int getPlayerInput(){
-        int number;
-        std::cout << "Please enter the number between 1 - 10 range:";
-        while(std::cin >> number && !(number >= 1 && number <= 10)){
+
+    Character getPlayerInput(Character &humCh){
```

Comment:
I would like that we create new entity
Input {
number1
number2
}

Also that we have 1 method for getNumber from console.
it will look like
getInputFromConsole{
int number1 = getNumberFromConsole()
int number2
while(number1 != number2) {
number2 = getNumberFromConsole()
}

and one method for getting number fron colsole.
And validate in this methos for get only number, and avoid that user input 'dsjfhjksdhfkjdsf!@3434' only number
[1 to 10]